### PR TITLE
Send reset password email endpoint

### LINKED ISSUE #507: Implement send reset password email endpoint
Implement reset password email endpoint that sends an email to the user which gives him access to a link to reset their password
This is a subtask of https://github.com/JamesPartsafas/ConcordiaCurriculumManager/issues/508

[Story Points: 2]
[Risk: High]
[Priority: High]

### PR DESCRIPTION
**Pull request by zihao429**
This PR closes #507

### CODE REVIEW

File: Server/ConcordiaCurriculumManager/appsettings.Development.json [hunk 1]
```diff
@@ -21,15 +21,34 @@
   "SenderEmailSettings": {
     "SenderSMTPHost": "smtp.gmail.com",
     "SenderSMTPPort": "587",
-    "SenderEmail": "test@gmail.com",
-    "SenderPassword": "not a valid password"
+    "SenderEmail": "ccmc9531@gmail.com",
+    "SenderPassword": "fgym vpwn eths umii"
```
Comment (JamesPartsafas) [inline Server/ConcordiaCurriculumManager/appsettings.Development.json:25]: If this email is linked to your phone number, you may want to remove this from version control. It's up to you.

File: Server/ConcordiaCurriculumManager/appsettings.Development.json [hunk 2]
```diff
@@ -21,15 +21,34 @@
...
+    "MinimumLevel": {
+      "Default": "Information",
+      "Override": {
+        "Microsoft": "Warning"
+      }
+    },
     "WriteTo": [
       {
-        "Name": "Console",
+        "Name": "PostgreSQL",
```
Comment (JamesPartsafas) [inline Server/ConcordiaCurriculumManager/appsettings.Development.json:36]: Is there a reason you change all this config here?

**Review by JamesPartsafas — APPROVED** [review]
Looks good, just some concerns with the config file that might need to be addressed

File: Server/ConcordiaCurriculumManager/appsettings.Development.json (hunk 2 above)
Comment (zihao429) [reply Server/ConcordiaCurriculumManager/appsettings.Development.json:36]: I didn't change that

File: Server/ConcordiaCurriculumManager/appsettings.Development.json (hunk 1 above)
Comment (zihao429) [reply Server/ConcordiaCurriculumManager/appsettings.Development.json:25]: Ok, ill take it off

**Pull request merged**